### PR TITLE
#60 FUNCTIONAL PROGRAMMING

### PR DESCRIPTION
**Pull request by ivelin980**

### CODE REVIEW

File: JavaAdvanced_functionalProgramming/src/jv/advanced/lab/functional/programming/SumNumbers.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * @since 24.12.2020
+ *
+ */
+public class SumNumbers {
+
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+		int[] numbers = Arrays.stream(scan.nextLine().split(", ")).mapToInt(Integer::parseInt).toArray();
+		Function<int[], Void> printCount = arr -> {
+			System.out.println("Count = " + arr.length);
```
Comment (ngavrailov) [inline JavaAdvanced_functionalProgramming/src/jv/advanced/lab/functional/programming/SumNumbers.java:21]: Why you did it in a different way for count and sum?

**Review by ngavrailov — CHANGES_REQUESTED** [review]

File: JavaAdvanced_functionalProgramming/src/jv/advanced/lab/functional/programming/SumNumbers.java (hunk 1 above)
Comment (ivelin980) [reply JavaAdvanced_functionalProgramming/src/jv/advanced/lab/functional/programming/SumNumbers.java:21]: To show that there are 2 different ways to use it.

File: JavaAdvanced_functionalProgramming/src/jv/advanced/exercise/functional/programming/ReverseAndExclude.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * 
+ * @author Ivelin Gavrailov
+ * @since 26.12.2020
+ *
+ */
+public class ReverseAndExclude {
+
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+		List<Integer> numbers = Arrays.stream(scan.nextLine().split("\\s+")).mapToInt(Integer::parseInt).boxed().collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply JavaAdvanced_functionalProgramming/src/jv/advanced/exercise/functional/programming/ReverseAndExclude.java:22]: fixed

File: JavaAdvanced_functionalProgramming/src/jv/advanced/exercise/functional/programming/CustomComparator.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * @since 27.12.2020
+ *
+ */
+public class CustomComparator {
+
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+		Arrays.stream(scan.nextLine().split("\\s+")).map(Integer::parseInt).sorted((f,s) -> {
+			int result = Integer.compare(Math.abs(f % 2),Math.abs(s % 2));
+			if(result == 0) {
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply JavaAdvanced_functionalProgramming/src/jv/advanced/exercise/functional/programming/CustomComparator.java:20]: fixed

File: JavaAdvanced_functionalProgramming/src/jv/advanced/exercise/functional/programming/AppliedArithmetic.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+			return list;
+		});
+		unaryMap.put("subtract", list -> {
+			for (int i = 0; i < list.size(); i++) {
+				numbers.set(i, numbers.get(i)-1);
+			}
+			return list;
+		});
+		while (!"end".equalsIgnoreCase(action)) {
+		if("print".equals(action)) {
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply JavaAdvanced_functionalProgramming/src/jv/advanced/exercise/functional/programming/AppliedArithmetic.java:50]: fixed

**Review by ngavrailov — APPROVED** [review]

**Pull request merged**